### ISSUE TITLE
Getting started: Pundit Scopes

### ISSUE THREAD

**Issue by MincePie** [elabs/pundit#437]
Sorry to make an issue request here. I'm out of luck in looking for places to try to find help.

I've wasted thousands of dollars on codementor.io in trying to learn how to use pundit, spent that much again on private consultants to try to teach me and spent hours trawling GitHub for examples. I've also posted at least 20 requests for help on StackOverflow to try to figure out how to get started with Pundit.

I've outlined my issue in these two posts. I've tried all of the things I've listed in these posts as well as a million others.

If you know how to use Pundit (with scopes) and are open to paid consulting, I would very much appreciate **expert**  assistance in find a way to get started.

http://stackoverflow.com/questions/41094763/rails-5-pundit-scopes
http://stackoverflow.com/questions/39186881/rails-4-pundit-scoped-policy-for-index?rq=1

In short, my issue is I can't make sense of the readme, the tutorials or any of the many posts on using this gem. I'm trying to make a scope so that I can render different records in an index depending on who the current_user is. Nothing I've been trying for the last 3 years has worked.

**Comment by rystraum** [comment]
Hi @MincePie,

I looked over your Rails 5 post in StackOverflow and your primary issue (undefined method `proposals` for nil:NilClass) should be fixed by replacing `@user` with `current_user`.

Also, adding additional context for the scope is called more manually:

In my own code, I have the following Scope:

```
class ReviewPolicy < ApplicationPolicy
  class Scope < Scope
    attr_reader :params

    def initialize(user, scope, params = {})
      super(user, scope)
      @params = params
    end

    def resolve
      listing = Listing.find_by(id: params[:listing_id])
      
      if listing.nil?
        return user.reviews if user
        return Review.none
      end

      return listing.reviews if user == listing.user
      return Review.approved_reviews(listing.id).or(Review.where(user_id: user.id)) if user
      return listing.approved_reviews
    end
  end
end
```

In the controller, I call it manually (i.e. not using Pundit helpers):
`ReviewPolicy::Scope.new(current_user, Review, params).resolve`
